From 45c5f7400f950b7f7ed0b7cdfd982acc9cac0cf2 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Mon, 7 Sep 2015 22:23:01 +0800
Subject: [PATCH] MLK-11511 ARM: imx: force ARM into wfi during LPSR enter

During stress test of LPSR using RTC alarm, sometimes
RTC alarm fail to fire, the control register(offset 0x38)
is changed and RTC alarm is disabled, according to
design team, when enter LPSR with programming SNVS
to shutdown PMIC, SNVS need some time to finish power
down process, better to put ARM into wfi during this
window.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-imx/suspend-imx7.S | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-imx/suspend-imx7.S b/arch/arm/mach-imx/suspend-imx7.S
index 885008c..34910ad 100644
--- a/arch/arm/mach-imx/suspend-imx7.S
+++ b/arch/arm/mach-imx/suspend-imx7.S
@@ -512,6 +512,11 @@ ENTRY(imx7_suspend)
 	orr	r7, r7, #0x60
 	str	r7, [r11, #0x38]
 wait_shutdown:
+	wfi
+	nop
+	nop
+	nop
+	nop
 	b	wait_shutdown
 
 ddr_only_self_refresh:
-- 
1.8.0

